### PR TITLE
[RSDK-7221] relax pinned versions of rustls

### PR DESCRIPTION
**Pull request by npmenard**
This PR updates rustls and viam-rust-utils allowing to relax the pinned version of futures-rustls

### CODE REVIEW

**Review by mattjperez — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ esp-idf-svc = { version = "=0.48.1", default-features = false }
 espflash = { git = "https://github.com/viamrobotics/espflash.git", branch = "monitor-output" }
 futures = "0.3.28"
 futures-lite = "1"
-futures-rustls = "=0.22.0"
+futures-rustls = "0.24"
```
Comment (acmorrow) [inline Cargo.toml:60]: Just curious why 0.24 and not the latest which is 0.26.

File: Cargo.toml [hunk 2]
```diff
@@ -86,7 +86,7 @@ reqwest = "0.11.20"
 reqwless = "0.5.0"
 ring = "0.16.20"
 ringbuf = "0.3.3"
-rustls = { version = "0.20.7", features = ["logging", "tls12"] }
+rustls = { version = "0.21", features = ["logging", "tls12"] }
```
Comment (acmorrow) [inline Cargo.toml:89]: Ditto why not 0.23 which is latest.

**Review by acmorrow — APPROVED** [review]
LGTM, though I'd like to know why we are upgrading to not-quite-latest. A fine answer to those questions is "because it doesn't work to do so".

File: Cargo.toml (hunk 1 above)
Comment (npmenard) [reply Cargo.toml:60]: it clashes with viam-rust-utils which is stuck in the past because of the bug with DTLS padding :P

File: Cargo.toml (hunk 2 above)
Comment (npmenard) [reply Cargo.toml:89]: same as above

**Pull request merged**